### DISCUSSION TITLE
Google PageSpeed Insights tells me to "Reduce initial server response time" for my Next.js app. How can I do that?

### DISCUSSION THREAD

**Discussion by lumenwrites** [vercel/next.js#28801] [Help]
It shows estimated savings as 1.22 seconds, which seems significant. 

What could be causing the issue? How can I improve this?

My app is [here](https://godotassets.io/). The code is [here](https://github.com/lumenwrites/godotassets). I've just started working on it, it barely does anything, just login functionality, so I don't think I can have any sort of complicated heavy code in there.

If that's relevant:

- My app is hosted on Vercel. 
- When I load a page it runs a serverless function that fetches data from the database hosted with AWS RDS.
- When I build the app, all the numbers in "First Load JS" are green (all the js chunks are under 100kb).

**Comment by erikdstock** [comment]
Assuming your only goal is to bring that number down and the time is being spent in a serverless function on initial request, can you render a spinner and load session functionality via xhr from the client? the `useSWR` hook may be helpful here.

**Reply by lumenwrites** [reply]
Unfortunately that would defeat the purpose of server-side rendering, I want all the posts in my app to be easily indexable by search engines.
